packages/flutter/lib/src/widgets/selectable_region.dart

```diff
@@ -490,11 +490,40 @@ class SelectableRegionState extends State<SelectableRegion> with TextSelectionDe
   }
 
   void _handleRightClickDown(TapDownDetails details) {
+    final Offset? previousSecondaryTapDownPosition = lastSecondaryTapDownPosition;
+    final bool toolbarIsVisible = _selectionOverlay?.toolbarIsVisible ?? false;
     lastSecondaryTapDownPosition = details.globalPosition;
     widget.focusNode.requestFocus();
-    _selectWordAt(offset: details.globalPosition);
-    _showHandles();
-    _showToolbar(location: details.globalPosition);
+    switch (defaultTargetPlatform) {
+      case TargetPlatform.android:
+      case TargetPlatform.fuchsia:
+      case TargetPlatform.windows:
+        _selectStartTo(offset: details.globalPosition);
+        _selectEndTo(offset: details.globalPosition);
+        _showHandles();
+        _showToolbar(location: details.globalPosition);
+      case TargetPlatform.iOS:
+        _selectWordAt(offset: details.globalPosition);
+        _showHandles();
+        _showToolbar(location: details.globalPosition);
+      case TargetPlatform.macOS:
+        if (previousSecondaryTapDownPosition == lastSecondaryTapDownPosition && toolbarIsVisible) {
+          hideToolbar();
+          return;
+        }
+        _selectWordAt(offset: details.globalPosition);
+        _showHandles();
+        _showToolbar(location: details.globalPosition);
+      case TargetPlatform.linux:
+        if (toolbarIsVisible) {
+          hideToolbar();
```

Comment:
is this true even if user click out side of the current selection?

Reply:
Yes, and it maintains the current selection.